packages/next/src/lib/helpers/get-registry.ts

```diff
@@ -8,22 +8,38 @@ import { getFormattedNodeOptionsWithoutInspect } from '../../server/lib/utils'
  * @default https://registry.npmjs.org/
  */
 export function getRegistry(baseDir: string = process.cwd()) {
+  const pkgManager = getPkgManager(baseDir)
+  // Since `npm config` command fails in npm workspace to prevent workspace config conflicts,
+  // add `--no-workspaces` flag to run under the context of the root project only.
+  // Safe for non-workspace projects as it's equivalent to default `--workspaces=false`.
+  // x-ref: https://github.com/vercel/next.js/issues/47121#issuecomment-1499044345
+  // x-ref: https://github.com/npm/statusboard/issues/371#issue-920669998
+  const resolvedFlags = pkgManager === 'npm' ? '--no-workspaces' : ''
   let registry = `https://registry.npmjs.org/`
+
   try {
-    const pkgManager = getPkgManager(baseDir)
-    const output = execSync(`${pkgManager} config get registry`, {
-      env: {
-        ...process.env,
-        NODE_OPTIONS: getFormattedNodeOptionsWithoutInspect(),
-      },
-    })
+    const output = execSync(
+      `${pkgManager} config get registry ${resolvedFlags}`,
+      {
+        env: {
+          ...process.env,
+          NODE_OPTIONS: getFormattedNodeOptionsWithoutInspect(),
+        },
+      }
+    )
       .toString()
       .trim()
 
     if (output.startsWith('http')) {
       registry = output.endsWith('/') ? output : `${output}/`
     }
-  } finally {
-    return registry
+  } catch (err) {
+    if (err instanceof Error) {
+      throw new Error(
+        `Failed to get registry from "${pkgManager}".\n\n${err.message}`
+      )
+    }
```

Comment:
Can we start using `cause` i.e. `throw new Error(`your message here`, { cause: error })`? This'll also make sure we handle the cases where non-errors are thrown.